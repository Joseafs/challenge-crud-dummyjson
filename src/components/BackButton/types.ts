import { Props as ButtonProps } from '~/components/Button/types';

export interface Props extends Omit<ButtonProps, 'children'> {}
