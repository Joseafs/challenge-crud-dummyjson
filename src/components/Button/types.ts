import { ButtonHTMLAttributes, MouseEventHandler, PropsWithChildren } from 'react';

import { PropsThemePaletteKeys } from '~/theme/config/types';

export type PropsButtonColor = {
  color?: PropsThemePaletteKeys;
  fullWidth?: boolean;
};

export interface Props extends PropsWithChildren, PropsButtonColor {
  className?: string;
  disabled?: boolean;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  text?: string;
  title?: string;
  type?: ButtonHTMLAttributes<undefined>['type'];
}
