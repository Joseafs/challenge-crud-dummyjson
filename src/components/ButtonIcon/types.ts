import { IconType } from 'react-icons';

import { Props as ButtonProps } from '~/components/Button/types';

export interface Props extends Omit<ButtonProps, 'children'> {
  icon: IconType;
}
