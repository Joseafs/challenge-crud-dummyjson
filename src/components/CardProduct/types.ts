import { MouseEventHandler } from 'react';

import { Product } from '~/services/products/getProducts/types';

type onClick = MouseEventHandler<HTMLButtonElement>;

export interface Props {
  className?: string;
  onDelete: onClick;
  onEdit: onClick;
  product: Product;
}
