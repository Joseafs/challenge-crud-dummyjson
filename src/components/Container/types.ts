import { PropsWithChildren } from 'react';

export type StyleRootProps = {
  className?: string;
  fixed?: boolean;
};

export interface Props extends PropsWithChildren<StyleRootProps> {}
