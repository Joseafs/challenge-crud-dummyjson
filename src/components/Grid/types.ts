import { ReactNode } from 'react';

import { PropsThemeColors } from '~/theme/config/types';

export type PropsGrid = {
  align?: 'left' | 'center' | 'right';
  backgroundColor?: PropsThemeColors;
  flex?: boolean;
  margin?: number[];
  overflow?: 'visible' | 'hidden' | 'scroll' | 'auto';
  padding?: number[];
  zIndex?: string;
};

export interface Props extends PropsGrid {
  children?: ReactNode;
  className?: string;
}
