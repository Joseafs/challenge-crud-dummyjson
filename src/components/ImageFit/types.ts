import Image from 'next/image';

export type PropsObjectFit = 'cover' | 'contain';

export type PropsImageFit = {
  fit?: PropsObjectFit;
};

type ImageNext = typeof Image;

export interface Props extends PropsImageFit, Partial<ImageNext> {
  className?: string;
  description?: string;
  fill?: boolean;
  height?: number;
  src: string;
  width?: number;
}
