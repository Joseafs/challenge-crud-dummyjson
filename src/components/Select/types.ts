import { ChangeEvent } from 'react';

export interface SelectOptions<Value = string> {
  label: string;
  value: Value;
}

export interface SelectProps {
  className?: string;
  label?: string;
  name?: string;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  options: { label: string; value: string }[];
  value?: string;
}
