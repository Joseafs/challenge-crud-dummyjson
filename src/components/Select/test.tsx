import { faker } from '@faker-js/faker';
import { fireEvent, render } from '@testing-library/react';

import { withThemeProvider } from '~/theme/utils';
import { generateArray } from '~/utils/generateArray';
import { snapshotSelectValues } from '~/utils/mocks/snapshotMockValues';

import { SelectOptions } from './types';

import { Select } from '.';

describe('Select', () => {
  test('Should match snapshot', () => {
    const mockFn = jest.fn();

    const { container } = render(
      withThemeProvider(<Select label="select-field" name="faker" onChange={mockFn} options={snapshotSelectValues} />),
    );

    expect(container).toMatchSnapshot();
  });

  test('Should change select to new value', () => {
    const options = generateArray<SelectOptions>(4, () => ({
      label: faker.lorem.words(1),
      value: faker.lorem.words(1),
    }));

    const mockFn = jest.fn();
    const { getByTestId } = render(
      withThemeProvider(<Select label="select-field" name="faker" onChange={mockFn} options={options} />),
    );

    const newSelecteValue = options[2].value;

    const input = getByTestId('select-faker') as HTMLInputElement;
    fireEvent.change(input, { target: { value: newSelecteValue } });
    expect(input.value).toBe(newSelecteValue);
    expect(mockFn).toHaveBeenCalledTimes(1);
  });
});
