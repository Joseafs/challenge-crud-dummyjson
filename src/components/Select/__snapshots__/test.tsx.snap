// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Should match snapshot 1`] = `
<div>
  <div
    class="sc-217a1f4b-0 ifUWuy"
  >
    <label
      class="sc-217a1f4b-3 bciKkD"
    >
      <div
        class="sc-217a1f4b-4 eTzaEb"
      >
        select-field
      </div>
    </label>
    <select
      class="sc-217a1f4b-1 dRfoem"
      data-testid="select-faker"
    >
      <option
        class="sc-217a1f4b-2"
        value="itachi-uchiha"
      >
        Itachi Uchiha
      </option>
      <option
        class="sc-217a1f4b-2"
        value="tanjiro-kamado"
      >
        Tanjiro Kamado
      </option>
      <option
        class="sc-217a1f4b-2"
        value="gon-freecss"
      >
        Gon Freecss
      </option>
    </select>
  </div>
</div>
`;
